x-pack/plugins/lens/public/datasources/text_based/components/dimension_editor.tsx

```diff
@@ -122,6 +174,43 @@ export function TextBasedDimensionEditor(props: TextBasedDimensionEditorProps) {
           {props.dataSectionExtra}
         </div>
       )}
+      {!isFullscreen && (
+        <div className="lnsIndexPatternDimensionEditor--padded lnsIndexPatternDimensionEditor--collapseNext">
+          {selectedField && (
+            <EuiText
+              size="s"
+              css={css`
+                margin-bottom: ${euiTheme.size.base};
+              `}
+            >
+              <h4>
+                {i18n.translate('xpack.lens.indexPattern.dimensionEditor.headingAppearance', {
+                  defaultMessage: 'Appearance',
+                })}
+              </h4>
+            </EuiText>
+          )}
+          <>
+            {selectedField && (
+              <NameInput
+                value={selectedField.label || ''}
```

Comment:
This field is not populated by default with the actual name that appears on the chart.

Reply:
addressed in https://github.com/elastic/kibana/pull/201540/commits/8772e75ef5cd79a7ddab7cc63886d9d8323c3136

Reply:
This still not working as the formbased datasoruce. If not the same at least please put the `defaultValue` in place so it is shown when clearing the field and by default initially.
https://github.com/user-attachments/assets/0bfb8c2a-8a16-4303-a2af-bb1c3f68d9cb

Reply:
this is left out at the moment as we dont know the default value in that place of the code.
i'll address this later when form and text based sources get merged into one.